File: examples/java/clientServer/ServerLib.java
import java.io.IOException;
import java.io.InputStream;
import java.net.ServerSocket;
import java.net.Socket;

public class ServerLib {

    public static String makeReply(String name) {
        return "Hello " + name;
    }

    public static void handleClient(Socket conn) throws IOException {
        try (conn) {
            InputStream in = conn.getInputStream();
            byte[] buf = new byte[256];
            int total = 0, n;
            while ((n = in.read(buf, total, buf.length - total)) > 0) {
                total += n;
            }
            if (total > 0) {
                String reply = makeReply(new String(buf, 0, total));
                conn.getOutputStream().write(reply.getBytes());
            }
        }
    }

    public static void runServer(int port) throws IOException {
        try (ServerSocket ss = new ServerSocket(port)) {
            System.out.println("Server listening on port " + port);
            while (true) {
                handleClient(ss.accept());
            }
        }
    }
}

File: examples/go/clientServer/server_lib.go
package clientserver

import (
	"fmt"
	"io"
	"net"
)

func MakeReply(name string) string {
	return "Hello " + name
}

// HandleConn reads a name from conn, writes the reply, and closes it.
func HandleConn(conn net.Conn) {
	defer conn.Close()
	data, err := io.ReadAll(conn)
	if err != nil || len(data) == 0 {
		return
	}
	conn.Write([]byte(MakeReply(string(data))))
}

// RunServer listens on addr and serves clients indefinitely.
func RunServer(addr string) error {
	ln, err := net.Listen("tcp", addr)
	if err != nil {
		return err
	}
	defer ln.Close()
	fmt.Println("Server listening on", addr)
	for {
		conn, err := ln.Accept()
		if err != nil {
			continue
		}
		go HandleConn(conn)
	}
}
